### DISCUSSION TITLE
为什么在next.js里面使用AntDesign 总是出现页面闪烁 也就是先现实原生组件 然后在显示具有css的 实在不知道为什么有人能够解答一下吗

### DISCUSSION THREAD

**Discussion by 2010363329** [ant-design/ant-design#46011] [Q&A]
`"use client";

import React from "react";
import { Button } from 'antd';

export default function Page() {

  return (
    <>
      <Button type="primary" size="large">登录 / 注册</Button>
      <h1 style={{ color : 'green' }}>test</h1>
    </>
  );
}
`
这是我的代码 新手求教

**Comment by li-jia-nan** [comment]
https://ant.design/docs/react/use-with-next-cn

**Reply by 2010363329** [reply]
这么麻烦吗 我不仅仅是antdesign页面闪烁 我尝试用css in js写也闪烁 这个怎么解决呢

**Reply by 2010363329** [reply]
我除了注释第三步骤的import '@/globals.css';
其他一模一样 还是页面闪烁说为什么 所有代码全部复制的
能帮忙解答一下吗 还有没有其他原因

**Comment by zombieJ** [comment]
可以看一下官方文档，NextJS 和其他框架不一样，默认 SSR。所以你需要写 style extract 来做样式注水。

**Reply by 2010363329** [reply]
看next.js的文档 还是antdesign的文档 我文档都看完了 也没找到这个

**Reply by zombieJ** [reply]
> 看next.js的文档 还是antdesign的文档 我文档都看完了 也没找到这个

NextJS: https://nextjs.org/docs/app/building-your-application/styling/css-in-js
Ant Design: https://ant-design.antgroup.com/docs/react/use-with-next-cn

**Reply by 2010363329** [reply]
好的 感谢 我看看  之前没仔细看这两个

**Reply by 2010363329** [reply]
我除了注释第三步骤的import '@/globals.css';
其他一模一样 还是页面闪烁说为什么 所有代码全部复制的
能帮忙解答一下吗 还有没有其他原因

**Reply by WRCoding** [reply]
最后怎么解决的，我按照文档说的来，还是会闪烁
